docs/src/tutorial/minimal_working_compute.jl

```diff
@@ -0,0 +1,266 @@
+that should allow you to quickly run a compute shader on actual data. In short,
+we walk through the following steps:
+
+- Opening a device and finding good queues families and memory types
+- Allocating memory and buffers
+- Compiling a shader program and filling up the structures necessary to run it:
+  - specialization constants
+  - push constants
+  - descriptor sets and layouts
+- Making a command buffer and submitting it to the queue, efficiently running
+  the shader
+
+## Initialization
+
+=#
+
+using SwiftShader_jll # hide
+using Vulkan
+@set_driver :SwiftShader # hide
+
+instance = Instance([], [])
+
+# Take the first available physical device (you might check that it is an
+# actual GPU, using [`get_physical_device_properties`](@ref)).
+physical_device = first(unwrap(enumerate_physical_devices(instance)))
+
+# At this point, we need to choose a queue family index to use. For this
+# example, have a look at `vulkaninfo` command and pick the good queue manually
+# from the list of `VkQueueFamilyProperties` -- you want one that has
+# `QUEUE_COMPUTE` in the flags. In a production environment, you would use
+# [`get_physical_device_queue_family_properties`](@ref) to find a good index.
+qfam_idx = 0
+
+# Create a device object and make a queue for our purposes.
+device = Device(physical_device, [DeviceQueueCreateInfo(qfam_idx, [1.0])], [], [])
+
+# ## Allocating the memory
+#
+# Same way, you need to find a good memory type. Again, you can find a good one
+# using `vulkaninfo` or with [`get_physical_device_memory_properties`](@ref).
+# For compute, you want something that is both at the device (contains
+# `MEMORY_PROPERTY_DEVICE_LOCAL_BIT`) and visible from the host
+# (`..._HOST_VISIBLE_BIT`).
+memorytype_idx = 0
+
+# Let's create some data. We will work with 100 flimsy floats.
+data_items = 100
+mem_size = sizeof(Float32) * data_items
+
+# Allocate the memory of the correct type
+mem = unwrap(allocate_memory(device, mem_size, memorytype_idx))
+
+# Make a buffer that will be used to access the memory, and bind it to the
+# memory. (Memory allocations may be quite demanding, it is therefore often
+# better to allocate a single big chunk of memory, and create multiple buffers
+# that view it as smaller arrays.)
+buffer = unwrap(create_buffer(device, mem_size, BUFFER_USAGE_STORAGE_BUFFER_BIT, SHARING_MODE_EXCLUSIVE, [qfam_idx]))
+
+bind_buffer_memory(device, buffer, mem, 0)
+
+# ## Uploading the data to the device
+#
+# First, map the memory and get a pointer to it.
+memptr = map_memory(device, mem, 0, mem_size)
+
+# Here we force Julia to look at the mapped data as at the vector of
+# `Float32`s, so that we can access it easily.
+data = unsafe_wrap(Vector{Float32}, convert(Ptr{Float32}, unwrap(memptr)), data_items, own = false)
+
+# For now, let's just zero out all the data, and *flush* the changes to make
+# sure the device can see the updated data. This is the simplest way to move
+# array data to the device.
```

Comment:
```suggestion
# For now, let's just zero out all the data, and *flush* the changes to make
# sure the device can see the updated data. This is the simplest way to move
# array data to the device.
# Note that if the memory is host-coherent (`MEMORY_PROPERTY_HOST_COHERENT_BIT`),
# then writes don't have to be flushed.
```

Reply:
I added a bit of danger tone in this, forgetting to flush/invalidate is one source of bad errors. :D